spec/controllers/batch_controller_spec.rb

```diff
@@ -22,6 +22,23 @@
       end
     end
 
+    describe "when submiting files on behalf of another user" do
+      let(:somebody_else_file) do 
+        f = GenericFile.create(title: ['Original Title'])
+        f.apply_depositor_metadata('current_user')
+        f.save!
+        f.on_behalf_of = 'somebody@else.org'
```

Comment:
I couldn't figure out a way to set `on_behalf_of` without the extra save. Setting it as a parameter in the call to `create` or in between `apply_depositor_metadata` and `save` throws error `NoMethodError: undefined method `log_event' for nil:NilClass`